src/xdgiconloader/xdgiconloader.cpp

```diff
@@ -841,8 +841,8 @@ QPixmap ScalableFollowsColorEntry::pixmap(const QSize &size, QIcon::Mode mode, Q
             const QPalette pal = qApp->palette();
             // Note: indexes are assembled as in qtsvg (QSvgIconEnginePrivate::hashKey())
             QMap<int, QString> style_sheets;
-            style_sheets[(QIcon::Normal<<4)|QIcon::Off] = QStringLiteral(".ColorScheme-Text{color:%1;}").arg(pal.windowText().color().name());
-            style_sheets[(QIcon::Selected<<4)|QIcon::Off] = QStringLiteral(".ColorScheme-Text{color:%1;}").arg(pal.highlightedText().color().name());
+            style_sheets[(QIcon::Normal<<4)|QIcon::Off] = QStringLiteral(".ColorScheme-Text,.ColorScheme-PositiveText,.ColorScheme-NeutralText,.ColorScheme-NegativeText{color:%1;}").arg(pal.windowText().color().name());
```

Comment:
Why do we define the same color for all of the classes? Shouldn't we do it the way as kiconloader does it?
https://github.com/KDE/kiconthemes/blob/master/src/kiconloader.cpp#L75
https://github.com/KDE/kiconthemes/blob/master/src/kiconloader.cpp#L898

Reply:
Explained in the patch comment: those extra colors aren't Qt palette colors but KDE specific ones. Here, we just tolerate their names.